examples/http/vmware_vcenter_monitoring_system_health.js

```diff
@@ -0,0 +1,159 @@
+    };
+    D.device.http.post(config, function(error, response, body){
+        if (error) {          
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else if (response.statusCode == 404) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode == 401) {
+            D.failure(D.errorType.AUTHENTICATION_ERROR);
+        } else if (response.statusCode != 201) {
+            D.failure(D.errorType.GENERIC_ERROR);
+        } 
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+/**
+ * Function to perform HTTP GET request
+ * @param {string} url The URL to send the GET request to
+ * @param {object} sessionId The response from the login request, containing the session ID
+ * @returns A promise containing the body of the response
+ */
+function httpGet(url, sessionId) {
+    var d = D.q.defer();
+    var config = {
+        url: url,
+        protocol: "https",
+        jar: true,
+        rejectUnauthorized: false,
+        headers: {
+            "vmware-api-session-id": sessionId
+        }
+    };
+    D.device.http.get(config, function(error, response, body){
+        if (error) {          
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else if (response.statusCode == 404) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode != 200) {
+            console.log(response.body);
+            d.resolve("N/A");
+        } 
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+// Function to get URLs for system health information
+function getHealthSystemsUrls(url) {
+    return function(sessionId) {
+        return httpGet(url, sessionId);
+    };
+}
+
+// Function to retrieve system health information
+function getSystemHealthInfo(sessionId) {
+    return D.q.all([
+        getHealthSystemsUrls("/api/appliance/health/system")(sessionId), //Get overall health of system
+        getHealthSystemsUrls("/api/appliance/health/load")(sessionId), //Get load health
+        getHealthSystemsUrls("/api/appliance/health/mem")(sessionId), //Get memory health
+        getHealthSystemsUrls("/api/appliance/health/applmgmt")(sessionId), //Get health status of applmgmt services
+        getHealthSystemsUrls("/api/appliance/health/storage")(sessionId), //Get storage health
+        getHealthSystemsUrls("/api/appliance/health/swap")(sessionId), //Get swap health
+        getHealthSystemsUrls("/api/appliance/health/database-storage")(sessionId), //Get database storage health
+        getHealthSystemsUrls("/api/appliance/health/software-packages")(sessionId) //Get information on available software updates available in the remote vSphere Update Manager repository
+    ]);
+}
+
+// This function extracts data from the response and populates the custom table 
+function extractData(data) {
+    var recordIds = [
```

Comment:
make this as a global configuration variable
config = {id: "system-health", url: "/api/appliance/health/system" ...}
in getSystemHealthInfo build a list of promises based on this config then in extract data make the same code based on the iteration on the id of the configuration